algorithms/Steamroller.js

```diff
@@ -0,0 +1,20 @@
+// Flatten a nested array. You must account for varying levels of nesting.
+
+
+function steamrollArray(arr) {
+  // I'm a steamroller, baby
+  var flattenArray = [];
+  while(arr.length > 0){
+    if(Array.isArray(arr[0]) === false){
+     flattenArray.push(arr[0]);
+     arr.shift(0);
```

Comment:
This implementation mutates the array that was provided as a parameter to the function.  This may be an unintended side effect that a user of the function would not anticipated.  I'm not sure which approach was required by the spec (it's unfortunate that I can't see the project goal you're working on to check for myself - we should file a feature request with prrr :P).  

Unless the specification specifically allows you to mutate the parameter, I suggest an implementation that does not mutate the input.